File: allure-docker-nodejs-typescript-example/features/steps/steps.ts
import { Given, When, Then } from "cucumber"
import { expect } from 'chai'

Given('I\'m on a site', function () {

});

When('I enter {string} on the page', function (string) {

});

Then('I verify is {string}', function (status) {
    switch(status) {
        case "FAILED":
            expect.fail("FAILURE ON PURPOSE")
        break
        case "BROKEN":
            return 'pending'
    }
});

File: allure-docker-python-behave-example/features/steps/steps.py
from behave import given, when, then, step

@given(u'I\'m on a site')
def step_impl(context):
    print('first step')

@when(u'I enter {value} on the page')
def step_impl(context, value):
    print('second step ' + value)

@then(u'I verify is "{status}"')
def step_impl(context, status):
    print('third step' + status)
    if status == "FAILED":
        assert False, "FAILURE ON PURPOSE"
    if status == "BROKEN":
        raise NotImplementedError('NOT IMPLEMENTED')